### DISCUSSION TITLE
Trouble Reproducing the Adversarial Patch Notebook Results

### DISCUSSION THREAD

**Discussion by dominic-simon** [Trusted-AI/adversarial-robustness-toolbox#1383] [Q&A] [answered]
As the title states, I'm having issues with the Adversarial Patch Notebook, specifically the TensorFlowV2 notebook. The patch shown on the notebook looks like this:
![index](https://user-images.githubusercontent.com/84974463/139386686-ba7127a6-ceac-4c4b-8731-79eb36a16de3.png)

While mine ends up looking like this:
![13_patched](https://user-images.githubusercontent.com/84974463/139386825-5a8a11f2-d605-4648-8149-1187085bf58f.png)

I have tried running the code from the notebook on both Google Colab and on my own computer, but generate the above patch in both instances. I have also tried multiple different classes for the patch, but the rainbow of colors stays for each different different class. 

Am I somehow not copying something correctly? Is there an issue with the version of a relevant library? Is it something else? I'm really not sure, so any help is appreciated. Of course, I'll provide any necessary information to help get to the bottom of this upon request. 

Thank you in advance for the assistance!

**Comment by beat-buesser** [comment, marked as answer]
Hi @dominic-simon Thank you very much for using ART! Would you be able to share your Google Colab notebook?

**Reply by dominic-simon** [reply]
Hello! Thank you for the response. 

Here is the link to the notebook:
https://colab.research.google.com/drive/1_cV9LBi7fMNjkM6Nym_-LgojlfQp1c-D?usp=sharing

**Reply by beat-buesser** [reply]
Hi @dominic-simon I think our notebook needs an updated value for the learning rate. We have recently changed the attacks optimizer from SGD to Adam. Can you try again with `learning_rate=5` or `learning_rate=1` and maybe more iterations if needed?

**Reply by dominic-simon** [reply]
@beat-buesser Will do! I'll let you know how it goes.

**Reply by dominic-simon** [reply]
@beat-buesser 
Your suggestion worked! So far, a learning rate of 1 with a max iteration of 100, has yielded a patch closest to the one presented in the notebook.

Now I'm trying to configure the parameters to see if I can produce a patch that looks more like the ones presented in the paper describing this patch.

Thanks so much for your help.
